sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -446,64 +327,21 @@ public String getEntityPath() {
      * @throws IllegalArgumentException if {@code lockToken} is an empty value.
      * @see <a href="https://docs.microsoft.com/azure/service-bus-messaging/message-deferral">Message deferral</a>
      */
-    public Mono<Void> defer(String lockToken, Map<String, Object> propertiesToModify) {
-        return defer(lockToken, propertiesToModify, receiverOptions.getSessionId());
+    public Mono<Void> defer(ServiceBusReceivedMessage message, Map<String, Object> propertiesToModify) {
+        return updateDisposition(message, DispositionStatus.DEFERRED, null, null,
+            propertiesToModify, null);
     }
 
     /**
-     * Defers a {@link ServiceBusReceivedMessage message} using its lock token with modified message property. This will
+     * Defers a {@link ServiceBusReceivedMessage message} with modified message property. This will
      * move message into the deferred subqueue.
      *
-     * @param lockToken Lock token of the message.
+     * @param message The {@link ServiceBusReceivedMessage} message.
```

Comment:
The docs will be odd to read. It'll show up as `The ServiceBusReceivedMessage messge`.